### PR TITLE
Water bottles are drinkable potions

### LINKED ISSUE #4111: Water bottle should be drinkable
Client version:   1.12.2
Server OS:        Windows
Commit id:        f0b05e78f70fa883bfdc74568bd7bf914025306c

### Expected behavior
Holding a water bottle, right click, and drinks the water bottle, returns a glass bottle.

### Actual behavior
Throw it away and it breaks.

### Steps to reproduce the behavior
As described before.

### PR DESCRIPTION
**Pull request by bearbin**
Fixes #4111 

There seem to be some other issues with potions as well that will need some more intensive fixes, but this will fix the immediate issue in #4111.

### CODE REVIEW

**Comment by Bond-009** [comment]
I don't think that this is a good solution as there also are water splash potions.

P.S. the ref to the mcwiki in the comment is outdated and should be replaced by this one: https://minecraft.gamepedia.com/Java_Edition_data_values#Potions

**Comment by peterbell10** [comment]
Testing the "splash_bit" from the wiki page seems to work correctly:
```cpp
return ((a_ItemDamage & 0x4000) == 0);
```

**Comment by bearbin** [comment]
> I don't think that this is a good solution as there also are water splash potions.

This solution works as water splash potions have a data value != 0. However @peterbell10's suggestion looks to be better.

> P.S. the ref to the mcwiki in the comment is outdated and should be replaced by this one: https://minecraft.gamepedia.com/Java_Edition_data_values#Potions

Done

> Testing the "splash_bit" from the wiki page seems to work correctly:

This looks to be a good solution, but it's not exactly clear that a function `IsPotionDrinkable` should check "Is potion not a spash potion". Are there any potions that are neither drinkable nor splash potions?

**Comment by peterbell10** [comment]
I don't think there are any non-drinkable, non-splash potions.  Still, It might be better to put splash bit testing into it's own function `IsPotionSplash`.  Especially as the original issue caused splash potion creation so obviously `IsPotionDrinkable` is being used in this sense.

The code seems to use not drinkable to imply splash in other places as well:

https://github.com/cuberite/cuberite/blob/4dcff34fecde217efd27bae310029d951d5f5252/src/Entities/EntityEffect.cpp#L112

**Comment by bearbin** [comment]
> I don't think there are any non-drinkable, non-splash potions.

OK, well if that's the case then I don't see any point in making a change to the name of the function. It may not be ideal, but since MC does not currently make the distinction there's no point in us making that distinction. I think changing the comment text as I have done is sufficient.

**Comment by Bond-009** [comment]
There are lingering potions, but I don't know if they are counted as a splash potion.
Ref: https://minecraft.gamepedia.com/Lingering_Potion

**Comment by bearbin** [comment]
I think the lingering potions are just splash potions, at least with regard to the splash potion data bit.

**Comment by peterbell10** [comment]
>I don't see any point in making a change to the name of the function
I was actually meaning a separate function but I suppose it doesn't really matter.

File: src/Entities/EntityEffect.cpp [hunk 1]
```diff
@@ -125,9 +125,10 @@ int cEntityEffect::GetPotionEffectDuration(short a_ItemDamage)
 
 bool cEntityEffect::IsPotionDrinkable(short a_ItemDamage)
 {
-	// Drinkable potion if 13th lowest bit is set
-	// Ref.: https://minecraft.gamepedia.com/Potions#Data_value_table
-	return ((a_ItemDamage & 0x2000) != 0);
+	// Potions are drinkable if they are not splash potions.
+	// i.e. potions are drinkable if the 14th lowest bit is not set
+	// Ref.: https://minecraft.gamepedia.com/Java_Edition_data_values#.22Splash_potion.22_bit
+	return (a_ItemDamage & 0x4000) == 0;
```
Comment (peterbell10) [inline src/Entities/EntityEffect.cpp]: Isn't the convention to _always_ have brackets around comparisons? Like how it was before.

Comment (bearbin) [reply src/Entities/EntityEffect.cpp]: Yes...

**Comment by Pokechu22** [comment]
Do we really want to keep the old potion ID system?  It's a bit of a mess IMO (though if 1.8 is still supported, something like this is needed).

**Comment by bearbin** [comment]
@Pokechu22 It works, and there are higher priority enhancements to make, but if you or anybody else wanted to give replacing it a go that would be great :)

**Comment by 9caihezi** [comment]
Is lingering potion implemented? Is so we need a new issue for it. Currently lingering will become splash potion, and cannot be made by dragon breath.

**Comment by bearbin** [comment]
I don't believe that lingering potions are implemented, as they were only added to vanilla in 1.9. In any case, unless there is another issue about implementing them it would be a good idea to create one.

**Pull request merged**